src/plugins/intel_cpu/src/nodes/dft.cpp

```diff
@@ -404,24 +481,221 @@ void DFT::naiveDFT(float* data, size_t dataLength) const {
+        nextIterationBlockSize /= 2;
+        if (parallelize && blockSize >= 4 * elementsPerCacheLine) {
+            parallel_for(numBlocks, [&](const size_t block) {
+                blockIteration(block, 1, blockSize, nextIterationBlockSize);
+            });
+        } else {
+            blockIteration(0, numBlocks, blockSize, nextIterationBlockSize);
+        }
+
+        std::swap(inBuffer, outBuffer);
+    }
+
+    if (inverse) {
+        for (int64_t k = 0; k < dataLength; k++) {
+            inBuffer[k] /= nComplex;
+        }
+    }
+
+    return inBuffer;
+}
+
+void DFT::DFTJitExecutor::naiveDFT(float* data, size_t dataLength, bool inverse) const {
+    std::vector<float> outputBuffer(dataLength);
+    const size_t nComplex = dataLength / 2;
+    const auto& twiddles = twiddlesMapDFT.find(nComplex)->second;
+
+    parallel_for(nComplex, [&](size_t k) {
+        auto arg = jit_args_dft();
+
+        arg.src = data;
+        arg.dst = outputBuffer.data() + 2 * k;
+        arg.twiddles = twiddles.data() + 2 * k * nComplex;
+        arg.work_amount = nComplex;
+        arg.index = k;
+
+        (*dftKernel)(&arg);
+    });
+    cpu_memcpy(data, outputBuffer.data(), dataLength * sizeof(float));
+}
+
 bool DFT::created() const {
     return getType() == Type::DFT;
 }
 
-void DFT::createPrimitive() {}
+bool DFT::needPrepareParams() const {
+    return lastInverse != inverse;
+}
+
+void DFT::prepareParams() {
+    DFTKey key = {};
+
+    key.nodeAttrs.inverse = inverse;
+    key.nodeAttrs.hasDFT = false;
+    key.nodeAttrs.hasFFT = false;
+
+    axes = getAxes();
+    const auto outputShape = getChildEdgesAtPort(0)[0]->getMemory().getStaticDims();
+
+    for (size_t axis : axes) {
+        size_t nComplex = outputShape[axis];
+        if (!IsPowerOfTwo(nComplex)) {
+            key.nodeAttrs.hasDFT = true;
+        } else {
+            key.nodeAttrs.hasFFT = true;
+        }
+    }
+
+    auto buildExecutor = [&](const DFTKey& key) -> std::shared_ptr<DFTExecutor> {
+        std::shared_ptr<DFTExecutor> executor;
+        if (mayiuse(cpu::x64::sse41)) {
+            executor = std::make_shared<DFTJitExecutor>(key.nodeAttrs);
+        } else {
+            executor = std::make_shared<DFTRefExecutor>(key.nodeAttrs);
+        }
+        return executor;
+    };
+
+    auto cache = getRuntimeCache();
+    auto result = cache->getOrCreate(key, buildExecutor);
+    execPtr = result.first;
+
+    lastInverse = inverse;
+}
+
+std::vector<int32_t> DFT::getAxes() const {
+    auto axesEdge = getParentEdgeAt(AXES_INDEX);
+    const auto* axesStartPtr = reinterpret_cast<const int32_t*>(axesEdge->getMemoryPtr()->GetPtr());
+    auto axes = std::vector<int32_t>(axesStartPtr, axesStartPtr + axesEdge->getMemory().getStaticDims()[0]);
+    for (auto& axis : axes) {
+        if (axis < 0) {
+            axis += inputShape.size() - 1;
+        }
+    }
+    std::sort(axes.begin(), axes.end());
+    return axes;
+}
+
+void DFT::createPrimitive() {
```

Comment:
Why do you override this function?

Reply:
Probably, it was left after some experiments; I removed it